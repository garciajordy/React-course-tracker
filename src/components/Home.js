import React, { useEffect } from 'react';
import axios from 'axios';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import Registration from './auth/Registration';
import LoginPanel from './auth/Login';
import store from '../store/Store';
import logIn from '../actions/LoginAction';

const Home = ({ loggedInStatus }) => {
  const history = useHistory();
  function checkLoginStatus() {
    axios.get('http://localhost:3001/logged_in', { withCredentials: true }).then((response) => {
      if (response.data.logged_in) {
        store.dispatch(logIn(response.data.user));
        history.push('/dashboard');
      }
    }).catch((error) => error);
  }
  useEffect(() => {
    checkLoginStatus();
  });
  return (
    <div className="dashboard">
      <div className="header">
        CourseTrack.it
      </div>
      <h1>Home</h1>
      <h1>{ loggedInStatus }</h1>
      <Registration />
      <LoginPanel />
    </div>
  );
};

Home.propTypes = {
  loggedInStatus: PropTypes.bool.isRequired,
};
export default Home;
