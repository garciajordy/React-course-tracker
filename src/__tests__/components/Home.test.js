import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Home from '../../components/Home';
import store from '../../store/Store';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><Home /></Provider>);
  expect(asFragment(<Provider store={store}><Home /></Provider>)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Provider store={store}><Home /></Provider>);
  expect(screen.getByText('Home')).toBeTruthy();
});
