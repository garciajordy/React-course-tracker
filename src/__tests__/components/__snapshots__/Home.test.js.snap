// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="dashboard"
  >
    <div
      class="header"
    >
      CourseTrack.it
    </div>
    <h1>
      Home
    </h1>
    <h1 />
    <div>
      <form>
        <p>
           
        </p>
        <input
          name="email"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
        <input
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <input
          name="password_confirmation"
          placeholder="Password_confirmation"
          required=""
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
    <div>
      <form>
        <p>
           
        </p>
        <input
          name="email"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
        <input
          name="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
